scipy/integrate/_ivp/ivp.py

```diff
@@ -26,28 +27,18 @@ class OdeResult(OptimizeResult):
 
 
 def prepare_events(events):
-    """Standardize event functions and extract is_terminal and direction."""
+    """Standardize event functions and extract attributes."""
     if callable(events):
         events = (events,)
 
-    if events is not None:
-        is_terminal = np.empty(len(events), dtype=bool)
-        direction = np.empty(len(events))
-        for i, event in enumerate(events):
-            try:
-                is_terminal[i] = event.terminal
-            except AttributeError:
-                is_terminal[i] = False
-
-            try:
-                direction[i] = event.direction
-            except AttributeError:
-                direction[i] = 0
-    else:
-        is_terminal = None
-        direction = None
+    max_events = np.empty(len(events))
+    direction = np.empty(len(events))
+    for i, event in enumerate(events):
+        terminal = int(event.terminal) if hasattr(event, 'terminal') else np.inf
+        max_events[i] = terminal if terminal > 0 else np.inf
```

Comment:
Any reason for not raising an error here? If I passed nonsense I think I'd rather know about it rather than silently receive some other behaviour

Reply:
When I write new functions, I am pretty careful about this sort of thing. When I work on old functions - especially for minor work like this - I consider how it will affect the overall quality of the function's input validation and try to balance that against simplicity of the PR. Currently, for example:
- If `event.direction=None`, no events are detected.
- If `event.terminal = 'no'`, then the function terminates after the first event. 

No errors are raised and these behavior are surprising, so I didn't think treating negative values like 0 made the function's edges substantially rougher.

However, seeing now that the current behavior for `event.terminal = -1` is *also* to cause the function to terminate after the first event, I agree that we should raise the error instead of changing that behavior. I'll change it.

Reply:
Oops, I'll also replace the ternary and `hasattr` with `getattr`.